### PR TITLE
feat(HACBS-2304): pass ec task bundle ver to release plr

### PR DESCRIPTION
**Pull request by johnbieren**

### CODE REVIEW

File: controllers/release/adapter.go [hunk 1]
```diff
@@ -232,7 +233,8 @@ func (a *Adapter) EnsureReleaseIsProcessed() (reconciler.OperationResult, error)
 		}
 
 		if pipelineRun == nil {
-			pipelineRun, err = a.createReleasePipelineRun(resources.ReleaseStrategy, resources.EnterpriseContractPolicy, resources.Snapshot)
+			pipelineRun, err = a.createReleasePipelineRun(resources.ReleaseStrategy, resources.EnterpriseContractConfigMap,
+				resources.EnterpriseContractPolicy, resources.Snapshot)
```
Comment (davidmogar) [inline controllers/release/adapter.go:237]: Let's pass the resources struct directly. It's becoming big.

File: tekton/pipeline_run.go [hunk 2]
```diff
@@ -69,6 +70,20 @@ func (r *ReleasePipelineRun) AsPipelineRun() *tektonv1beta1.PipelineRun {
 	return &r.PipelineRun
 }
 
+// WithEnterpriseContractConfigMap adds a param providing the verify ec task bundle to the release PipelineRun.
+func (r *ReleasePipelineRun) WithEnterpriseContractConfigMap(cm *corev1.ConfigMap) *ReleasePipelineRun {
```
Comment (davidmogar) [inline tekton/pipeline_run.go:74]: Rename to `configMap`. We are not in a letters shortage.

File: loader/loader.go [hunk 3]
```diff
@@ -129,6 +132,21 @@ func (l *loader) GetEnterpriseContractPolicy(ctx context.Context, cli client.Cli
 	return enterpriseContractPolicy, getObject(releaseStrategy.Spec.Policy, releaseStrategy.Namespace, cli, ctx, enterpriseContractPolicy)
 }
 
+// GetEnterpriseContractConfigMap returns the defaults ConfigMap in the Enterprise Contract namespace . If the EC_CM_NAMESPACEDNAME
+// value is invalid or not set, nil is returned. If the ConfigMap is not found or the Get operation fails, an error is returned.
+func (l *loader) GetEnterpriseContractConfigMap(ctx context.Context, cli client.Client) (*corev1.ConfigMap, error) {
+	enterpriseContractConfigMap := &corev1.ConfigMap{}
+	namespacedName := os.Getenv("EC_CM_NAMESPACEDNAME")
```
Comment (davidmogar) [inline loader/loader.go:139]: `ENTERPRISE_CONTRACT_CONFIG_MAP` sounds like a better name for the variable to me.

File: tekton/pipeline_run.go [hunk 4]
```diff
@@ -69,6 +70,20 @@ func (r *ReleasePipelineRun) AsPipelineRun() *tektonv1beta1.PipelineRun {
...
+	enterpriseContractConfigMapBundleField := "verify_ec_task_bundle"
+
+	ecTaskBundle := cm.Data[enterpriseContractConfigMapBundleField]
```
[the remark this answers is not in the document]
Comment (johnbieren) [reply tekton/pipeline_run.go:77]: https://stackoverflow.com/a/2050629 tells me that we would already use the empty string with the code written how it is if the item is not in the map

[the remark this answers is not in the document]
Comment (johnbieren) [reply tekton/pipeline_run.go:77]: Or do you mean if `Data` doesn't exist?

File: controllers/release/adapter.go (hunk 1 above)
Comment (johnbieren) [reply controllers/release/adapter.go]: done

File: tekton/pipeline_run.go (hunk 2 above)
Comment (johnbieren) [reply tekton/pipeline_run.go]: done

File: loader/loader_test.go [hunk 5]
```diff
@@ -281,13 +301,15 @@ var _ = Describe("Release Adapter", Ordered, func() {
 
 	Context("When calling GetProcessingResources", func() {
 		It("returns all the relevant resources", func() {
+			os.Setenv("ENTERPRISE_CONTRACT_CONFIG_MAP", "default/ec-defaults")
```
Comment (kasemAlem) [inline loader/loader_test.go:304]: should handle if the assignment failed for any reason ? like in  [ main.go Line137](https://github.com/redhat-appstudio/release-service/pull/205/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R137)

**Review by davidmogar — APPROVED** [review]

**Pull request merged**